src/Http/Client.php

```diff
@@ -159,10 +159,14 @@ class Client
      */
     public function __construct($config = [])
     {
+        if (!isset($config['adapter'])) {
+            $adapter = $this->_defaultConfig['adapter'];
+        } else {
+            $adapter = $config['adapter'];
+        }
+        $this->setConfig('adapter', $adapter);
```

Comment:
This is incorrect, once the adapter is extract it's supposed to be unset from config (set to null) like it was before.